wasm/tools/fixtures/runner.py

```diff
@@ -0,0 +1,443 @@
+
+    if ret == 'exhaustion':
+        raise Exhaustion("Exhaustion happened")
+    return ret
+
+
+def run_opcode_action_invoke(action, store, module, all_modules, registered_modules):
+    # get function name, which could include unicode bytes like \u001b which
+    # must be converted to unicode string
+    funcname = action.field
+    # print("funcname",funcname)
+    idx = 0
+    utf8_bytes = bytearray()
+    for c in funcname:
+        utf8_bytes += bytearray([ord(c)])
+    utf8_bytes = wasm.spec_binary_uN_inv(len(funcname), 32) + utf8_bytes
+    _, funcname = wasm.spec_binary_name(utf8_bytes, 0)
+    # print("funcname",funcname)
+    # get function address
+    funcaddr = None
+    # print(module["exports"])
+    # print("module",module)
+    # print("ok module",module)
+    # print("test ok",test)
+    for export in module["exports"]:
+        # print("export[\"name\"]",export["name"])
+        if export["name"] == funcname:
+            funcaddr = export["value"][1]
+    logger.debug("funcaddr: %s", funcaddr)
+
+    # print("funcaddr",funcaddr)
+    # funcbody = store["funcs"][funcaddr]["code"]["body"]
+    # print(wasm.print_tree_expr(funcbody))
+    # get args
+    args = []
+
+    for idx, arg in enumerate(action.args):
+        type_ = arg.type
+        value = arg.value
+
+        if type_ in {"f32", "f64"}:
+            logger.info("Floating point not yet supported: %s", action)
+            raise NotImplementedError("Floating point not yet implemented")
+        args += [[type_ + ".const", value]]
+
+    # invoke func
+    _, ret = wasm.invoke_func(store, funcaddr, args)
+
+    if ret == 'trap':
+        raise TrapError("TRAP")
+
+    return ret
+
+
+def run_opcode_action_get(action, store, module, all_modules, registered_modules):
+    exports = module["exports"]
+    # this is naive, since test["expected"] is a list, should iterate over each
+    # one, but maybe OK since there is only one test["action"]
+    for export in exports:
+        if export["name"] == action.field:
+            globaladdr = export["value"][1]
+            # print("store",store)
+            # print("store[\"gloabls\"]",store["globals"])
+            value = store["globals"][globaladdr]["value"][1]
+            # print("test_opcode_action_get",value)
+            return [value]
+            # num_tests_tried+=1
+            # if value != test["expected"][0]["value"]:
+            #  if verbose>1: print("SUCCESS")
+            # num_tests_passed+=1
+            # else:
+            #  if verbose>1: print("FAILURE")
+    raise Exception("Error?")
+
+
+def do_assert_return(command, store, module, all_modules, registered_modules):
+    try:
+        ret = run_opcode_action(command, store, module, all_modules, registered_modules)
+    except NotImplementedError:
+        return
+
+    if ret == "trap":
+        raise TrapError("Encountered TRAP while running command")
+
+    if len(ret) != len(command.expected):
```

Comment:
I have no idea what all the possible return values are for `ret` at this stage, but I think it can be either a handful of magic strings or a list of return values.